bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/handler/HydrawiseControllerHandler.java

```diff
@@ -0,0 +1,430 @@
+    // All responses use US local time formats
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("EEE, dd MMM uu HH:mm:ss Z",
+            Locale.US);
+    private final Map<String, @Nullable State> stateMap = Collections
+            .synchronizedMap(new HashMap<String, @Nullable State>());
+    private final Map<String, @Nullable Zone> zoneMaps = Collections
+            .synchronizedMap(new HashMap<String, @Nullable Zone>());
+    private int controllerId;
+
+    public HydrawiseControllerHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        HydrawiseControllerConfiguration config = getConfigAs(HydrawiseControllerConfiguration.class);
+        controllerId = config.controllerId;
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            HydrawiseAccountHandler handler = (HydrawiseAccountHandler) bridge.getHandler();
+            if (handler != null) {
+                handler.addControllerListeners(this);
+                if (bridge.getStatus() == ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand channel {} Command {}", channelUID.getAsString(), command.toFullString());
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.debug("Controller is NOT ONLINE and is not responding to commands");
+            return;
+        }
+
+        // remove our cached state for this, will be safely updated on next poll
+        stateMap.remove(channelUID.getAsString());
+
+        if (command instanceof RefreshType) {
+            // we already removed this from the cache
+            return;
+        }
+
+        HydrawiseGraphQLClient client = apiClient();
+        if (client == null) {
+            logger.debug("API client not found");
+            return;
+        }
+
+        String group = channelUID.getGroupId();
+        String channelId = channelUID.getIdWithoutGroup();
+        boolean allCommand = CHANNEL_GROUP_ALLZONES.equals(group);
+        Zone zone = zoneMaps.get(group);
+
+        if (!allCommand && zone == null) {
+            logger.debug("Zone not found {}", group);
+            return;
+        }
+
+        try {
+            switch (channelId) {
+                case CHANNEL_ZONE_RUN_CUSTOM:
+                    if (!(command instanceof QuantityType<?>)) {
+                        logger.warn("Invalid command type for run custom {}", command.getClass().getName());
+                        return;
+                    }
+                    if (allCommand) {
+                        client.runAllRelays(controllerId, ((QuantityType<?>) command).intValue());
```

Comment:
You need to convert the value to the unit you expect with `toUnit()`.

Reply:
thanks for catching that.